pkgs/development/libraries/libgcrypt/default.nix

```diff
@@ -47,7 +47,7 @@ stdenv.mkDerivation rec {
   # aarch64
   configurePlatforms = [ "host" "build" ];
 
-  postConfigure = ''
+  postConfigure = lib.optionalString (!stdenv.hostPlatform.isMinGW) ''
```

Comment:
this seems kinda spicy so if there's a better alternative I'm happy to change this

Reply:
@Frontear do you happen to know the equivalent of `--noexecstack` for mingw?

Reply:
I believe `-z noexecstack` should be given to the linker, vs. --noexecstack being given to the assembler like right now. From what I remember this should be correct

Reply:
Link to aforementioned issue: https://github.com/openssl/openssl/issues/9922

It might be possible to use `-z noexecstack` at the linking stage instead. At least thats the only other entrypoint I can think of worth testing. If neither works its likely still very much unsupported

Reply:
Hmm setting LDFLAGS doesn't appear to have an effect, will have to play around with this

Reply:
I'm pretty sure you can pass the LDFLAGS in the configure flags.

Reply:
> I'm pretty sure you can pass the LDFLAGS in the configure flags

I tried that below, though I don't believe it has an effect, since even invalid flags don't throw an error?

> You should also be checking the target platform instead of the host because of how cross compiling works.

I believe the check is correct.

```
nix-repl> pkgsCross.mingwW64.stdenv.buildPlatform.system
"x86_64-linux"

nix-repl> pkgsCross.mingwW64.stdenv.hostPlatform.system  
"x86_64-windows"

nix-repl> pkgsCross.mingwW64.stdenv.targetPlatform.system
"x86_64-windows"
```

Reply:
> I tried that below, though I don't believe it has an effect, since even invalid flags don't throw an error?

Tried the NIX_LDFLAGS as well?

> I believe the check is correct.

Oh, I just remember that there's some cases where host platform will not equal to your target and that could cause issues. Is there any specific reason as to why the LDFLAGS needs to apply when the host is targeted for mingw or can target work as well?

Reply:
> Tried the NIX_LDFLAGS as well?

Didn't, I'll try that.

> there's some cases where host platform will not equal to your target

Hm I'll have to think about that one carefully

Reply:
You won't have to :p Just ran a little test in another devshell, `$LD -z noexecstack` failed with "unrecognized option '-z'"